vortex-dtype/proto/dtype.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package vortex.dtype;
+
+enum PType {
+  U8 = 0;
+  U16 = 1;
+  U32 = 2;
+  U64 = 3;
+  I8 = 4;
+  I16 = 5;
+  I32 = 6;
+  I64 = 7;
+  F16 = 8;
+  F32 = 9;
+  F64 = 10;
+}
+
+message Null {}
+
+message Bool {
+  bool nullable = 1;
+}
+
+message Primitive {
+  PType type = 1;
+  bool nullable = 2;
+}
+
+message Decimal {
+  uint32 precision = 1;
+  uint32 scale = 2;
+  bool nullable = 3;
+}
+
+message Utf8 {
+  bool nullable = 1;
+}
+
+message Binary {
+  bool nullable = 1;
+}
+
+message Struct {
+  repeated string names = 1;
+  repeated DType dtypes = 2;
+  bool nullable = 3;
+}
+
+message List {
+  DType element_type = 1;
+  bool nullable = 2;
+}
+
+message Extension {
```

Comment:
Do we want these messages nested inside the DType message?

Reply:
I don't get this comment - where else would you put it? Do you want to do union of DType and Extension?

Reply:
```
message DType {
  message Primitive {..}
  message Extension {..}
}
```

It's purely a name spacing thing

Reply:
oh, that's what you mean. I haven't touched proto for too long so I forgot this is a thing you can do. My general rule was nest the messages unless there was some broader applicability.

Reply:
Actually I've decided against this _because_ there is broader applicability, e.g. a Struct dtype being used as a schema (if we pull up nullability, but that's a PR for later)